lib/client/src/client/client.ts

```diff
@@ -112,6 +123,51 @@ export interface AuthInfo {
  * @template R The type to use for ranges (such as `vscode.Range` for VS Code extensions).
  */
 export interface Client<R extends Range> {
+    /**
+     * Get the capabilities returned by the configured providers.
+     *
+     * It does not continue to listen for changes, as {@link Client.capabilitiesChanges} does. Using
+     * {@link Client.capabilities} is simpler and does not require use of observables (with a library like
+     * RxJS), but it means that the client needs to manually poll for updated item kinds if freshness is
+     * important.
+     */
+    capabilities(
+        params: CapabilitiesParams,
+        providerUri?: string
+    ): Promise<EachWithProviderUri<CapabilitiesResult[]>>
+
+    /**
+     * Observe OpenCtx capabilities of the configured providers.
+     *
+     * The returned observable streams capabilities as they are received from the providers and continues
+     * passing along any updates until unsubscribed.
+     */
+    capabilitiesChanges(
+        params: CapabilitiesParams,
+        providerUri?: string
+    ): Observable<EachWithProviderUri<CapabilitiesResult[]>>
+
+    /**
+     * Get the candidate items returned by the configured providers.
+     *
+     * It does not continue to listen for changes, as {@link Client.mentionsChanges} does. Using
+     * {@link Client.Mentions} is simpler and does not require use of observables (with a library like
+     * RxJS), but it means that the client needs to manually poll for updated items if freshness is
+     * important.
+     */
+    mentions(params: MentionsParams, providerUri?: string): Promise<EachWithProviderUri<MentionsResult>>
```

Comment:
ItemsParams doesn't contain a providerUri, but the response from this does. Is the idea that code that uses this client extracts out the providerURI from the mention and then sets it in the arguments to `items`.

Reply:
The implementation of the client handles that. Neither the Provider nor the consumer of the client needs to manage that.

The providers are owned by Client, so it can assign the Uris.